File: Ladder11/Level_3/KuriyamaStones.java
import java.util.*;
import java.io.*;

public class KuriyamaStones {

	public static void main(String[] args) {

		helper();
	}

	public static void helper() {

		int n = (int)(input());
		long[] arr = new long[n];
		TakeInput(arr);
		long[] sorted = arr.clone();
		Arrays.sort(sorted);
		for (int i = 1; i < n; i++) {
			arr[i] += arr[i - 1];
			sorted[i] += sorted[i - 1];
		}

		int t = (int)input();
		while (t-- > 0) {

			int type = (int)input();
			int l = (int)input() - 1;
			int r = (int)input() - 1;

			if (type == 1) {
				if (l == 0) {
					print(arr[r] + "\n");
				} else {
					print(arr[r] - arr[l - 1] + "\n");
				}
			} else {
				if (l == 0) {
					print(sorted[r] + "\n");
				} else {
					print(sorted[r] - sorted[l - 1] + "\n");
				}
			}
		}



	}

	public static long input() {
		return scn.input();
	}

	public static String input(int... var) {
		return scn.input(1);
	}

	public static void TakeInput(int[] arr) {
		for (int i = 0; i < arr.length; i++)
			arr[i] = (int)scn.input();
	}

	public static void TakeInput(long[] arr) {
		for (int i = 0; i < arr.length; i++)
			arr[i] = scn.input();
	}

	public static void printArray(int[] arr) {
		for (int val : arr)
			print(val + " ");

		print("\n");
	}

	public static void printArray(long[] arr) {
		for (long val : arr)
			print(val + " ");

		print("\n");
	}

	public static <T> void print(T t) {
		System.out.print(t);
	}
	public static void print() {
		System.out.println();
	}

	static FastScanner scn = new FastScanner(System.in);

	private static class FastScanner {

		BufferedReader br;
		StringTokenizer st;

		FastScanner(InputStream stream) {
			try {
				br = new BufferedReader(new InputStreamReader(stream));
			} catch (Exception e) {
				e.printStackTrace();
			}
		}

		String input(int a) {
			while (st == null || !st.hasMoreTokens()) {
				try {
					st = new StringTokenizer(br.readLine());
				} catch (IOException e) {
					e.printStackTrace();
				}
			}
			return st.nextToken();
		}

		long input() {
			return Long.parseLong(input(1));
		}

	}
}

File: Ladder11/Level_3/KuriyamaStones.cpp
#include <iostream>
#include <algorithm>
using namespace std;
long long arr[100001];
long long sorted[100001];

int main() {

	int n;
	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> arr[i];
		sorted[i] = arr[i];
	}

	sort(sorted, sorted + n);

	for (int i = 1; i < n; i++) {
		arr[i] += arr[i - 1];
		sorted[i] += sorted[i - 1];
	}

	int t ;
	cin >> t;
	while (t--) {

		int type, l, r;
		cin >> type >> l >> r;
		r--;
		l--;
		if (type == 1) {
			if (l == 0) {
				cout << arr[r] << endl;
			} else {
				cout << arr[r] - arr[l - 1] << endl;
			}
		} else {
			if (l == 0) {
				cout << sorted[r] << endl;
			} else {
				cout << sorted[r] - sorted[l - 1] << endl;
			}
		}
	}

}